### PR TITLE
[upload] - fix issue with uploading indicatorFields

### PR DESCRIPTION
**Pull request by GuyAfik**
## Related Issues
fixes: https://jira-hq.paloaltonetworks.local/browse/XSUP-28276

## Description
1. Fixed an issue where the body request when uploading single indicator fields was not correct
2. split the `IndicatorField` and `IncidentField` field into a new class `IndicatorIncidentField` because some of the functionality of them is identical (especially upload)

### CODE REVIEW

File: demisto_sdk/commands/content_graph/objects/indicator_incident_field.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
+from tempfile import NamedTemporaryFile
+
+import demisto_client
+from pydantic import Field
+
+from demisto_sdk.commands.common.constants import MarketplaceVersions
+from demisto_sdk.commands.common.handlers import JSON_Handler
+from demisto_sdk.commands.content_graph.objects.content_item import ContentItem
+
+json = JSON_Handler()
```
Comment (dorschw) [inline demisto_sdk/commands/content_graph/objects/indicator_incident_field.py:10]: use the DEFAULT_JSON_HANDLER

File: demisto_sdk/tests/integration_tests/upload_integration_test.py [hunk 2]
```diff
@@ -394,3 +394,51 @@ def test_upload_single_list(mocker, pack):
...
+            "fromVersion": "5.5.0",
+        },
+    )
+
+    mocker.patch(
+        "demisto_sdk.commands.upload.uploader.demisto_client", return_valure="object"
+    )
+    mocker.patch(
+        "demisto_sdk.commands.upload.uploader.get_demisto_version",
+        return_value=Version("6.8.0"),
```
[the remark this answers is not in the document]
Comment (GuyAfik) [reply demisto_sdk/tests/integration_tests/upload_integration_test.py:431]: we do

File: demisto_sdk/tests/integration_tests/upload_integration_test.py [hunk 3]
```diff
@@ -394,3 +394,51 @@ def test_upload_single_list(mocker, pack):
...
+    mocker.patch.object(ContentItem, "pack_version", return_value="1.2.0")
+
+    runner = CliRunner(mix_stderr=False)
+
+    with ChangeCWD(pack.repo_path):
+        result = runner.invoke(
+            main,
+            [UPLOAD_CMD, "-i", indicator_field.path],
+        )
+    assert result.exit_code == SUCCESS_RETURN_CODE
```
[the remark this answers is not in the document]
Comment (GuyAfik) [reply demisto_sdk/tests/integration_tests/upload_integration_test.py:444]: we send in only the `file name` to the http request and its being deleted during the execution so i can't know its content

what i can do is to make sure "SUCCESSFUL UPLOADS" is printed out making sure it was good.

File: demisto_sdk/commands/content_graph/objects/indicator_incident_field.py [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+
+class IndicatorIncidentField(ContentItem):
```
[the remark this answers is not in the document]
Comment (GuyAfik) [reply demisto_sdk/commands/content_graph/objects/indicator_incident_field.py:13]: This is similar to the `IntegrationScript` class which isn't abstract class.

[the remark this answers is not in the document]
Comment (dorschw) [reply demisto_sdk/commands/content_graph/objects/indicator_incident_field.py:12]: If making both doesn't require changing unit tests and passes build, let's do it

[the remark this answers is not in the document]
Comment (GuyAfik) [reply demisto_sdk/commands/content_graph/objects/indicator_incident_field.py:12]: looks like the `BaseContent` class already inherits the `ABC` so kinda irrelevant here

**Review by dorschw — APPROVED** [review]

**Pull request merged**